clients/src/test/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinatorTest.java

```diff
@@ -413,7 +413,7 @@ public void testJoinGroupRequestWithGroupInstanceIdNotFound() {
         assertTrue(consumerClient.poll(future, mockTime.timer(REQUEST_TIMEOUT_MS)));
         assertEquals(Errors.UNKNOWN_MEMBER_ID.message(), future.exception().getMessage());
         assertTrue(coordinator.rejoinNeededOrPending());
-        assertTrue(coordinator.hasMatchingGenerationId(defaultGeneration));
+        assertTrue(coordinator.hasUnknownGeneration());
```

Comment:
This test fix is piggy-backed, we should check that the generation is unknown instead of only testing the id itself is -1.